hydromt/workflows/basin_mask.py

```diff
@@ -173,12 +170,13 @@ def _parse_region_value(value):
         kwarg = dict(xy=value)
     elif isinstance(value, int):  # single int
         kwarg = dict(basid=value)
-    elif isinstance(value, (str, Path)) and isfile(value):
-        kwarg = dict(geom=gpd.read_file(value))
-    elif isinstance(value, gpd.GeoDataFrame):  # geometry
-        kwarg = dict(geom=value)
     elif isinstance(value, (str, Path)) and isdir(value):
         kwarg = dict(root=value)
+    elif isinstance(value, (str, Path)):
+        geom = data_catalog.get_geodataframe(value)
+        kwarg = dict(geom=geom)
+    elif isinstance(value, gpd.GeoDataFrame):  # geometry
+        kwarg = dict(geom=value)
```

Comment:
please add a new line, as the next `if` is presumably unrelated to this chain. I think it would also be good to add an `else` branch which raises a `ValueError` if someone passes in an unknown type